src/wallet/asyncrpcoperation_mergetoaddress.cpp

```diff
@@ -114,9 +115,6 @@ AsyncRPCOperation_mergetoaddress::AsyncRPCOperation_mergetoaddress(
     // Lock UTXOs
     lock_utxos();
     lock_notes();
-
-    // Enable payment disclosure if requested
-    paymentDisclosureMode = fExperimentalMode && GetBoolArg("-paymentdisclosure", false);
```

Comment:
I would keep this and set it to ``fExperimentalPaymentDislosure``. That approach makes it unit-testable without setting global variables.